django/contrib/gis/db/models/functions.py

```diff
@@ -190,11 +197,17 @@ def __init__(self, expression, num_seg=48, **extra):
 
 
 class Centroid(GeoFunc):
-    pass
+    def as_oracle(self, compiler, connection):
+        tol = self.extra.get('tolerance', 0.05)
+        self.template = "%%(function)s(%%(expressions)s, %s)" % tol
+        return super(Centroid, self).as_sql(compiler, connection)
```

Comment:
Definitely that would make more than sense. I'll implement it that before merging so it wouldn't be so repetitive code.